### ISSUE TITLE
 JACUSA_to_TRIBE.pl obsolete?

### ISSUE THREAD

**Issue by ckuenne** [dieterich-lab/JACUSA2#45]
Is the JACUSA_to_TRIBE.pl script found here (https://github.com/dieterich-lab/tribe-workflow/tree/master/RRD_workflow) still working for the JACUSA2 output format? Or is that functionality supposed to be superseded by JACUSA2helper? If so, can you suggest a workflow? Since TRIBE was one of the original selling points for JACUSA, an automation or tutorial for this specific pipeline might be of wider interest.

**Comment by piechottam** [comment]
Dear ckuenne, 

Thank you for your feedback and detailed problem description!


1. NullPointerException
    The SAM Output that you provided, suggests that STAR does not adhere to recommended practices for SAM format 
    https://samtools.github.io/hts-specs/SAMtags.pdf
    The required "NM" tag is not present in the SAM file.
    According to STAR manual (https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf):
    [...]
    The SAM attributes can be specified by the user using --outSAMattributes A1 A2 A3 ... option
    which accept a list of 2-character SAM attributes. The implemented attributes are: NH HI **NM** MD
    AS nM jM jI XS. **By default, STAR outputs NH HI AS nM attributes.**
    [...]
    I am afraid, you will need to rerun STAR with "--outSAMattributes NM NH [...and what ever you need...]"

    STAR output has the nonstandard field "nM".
    From SAMtags.pdf: [...] Note that tags starting with ‘X’, ‘Y’, or ‘Z’ and tags containing **lowercase** letters
    in either position are reserved for local use and will not be formally defined in any future version of this [...]

2. JACUSA_to_TRIBE.pl
    The person responsible for this PERL script is on vacation... as soon as he is back, I'll ask him to provide feedback.
    There are some workflows on https://dieterich-lab.github.io/JACUSA2helper/ (Articles).
    In general, we aim to add "useful" methods to JACUSA2helper.

**Comment by ckuenne** [comment]
Dear piechottam,

Thanks for the quick response! I have rerun STAR with "--outSAMattributes NH HI AS nM NM MD" and that seems to have solved the issue. JACUSA2 is currently running with -B A2G. I might be back with more feedback once that is done.

**Comment by ckuenne** [comment]
another question to JACUSA_to_TRIBE.pl: does it only work for stranded_forward or also for stranded_reverse libraries?
